test/command_line/suite/grndb/test_recover.rb

```diff
@@ -68,20 +82,30 @@ def test_default
       error = assert_raise(CommandRunner::Error) do
         grndb("recover")
       end
-      assert_equal(<<-MESSAGE, error.error_output)
+      messages = <<-MESSAGE
 Failed to recover database: <#{@database_path}>
 object corrupt: <[db][recover] column may be broken: <Users.age>: please truncate the column (or clear lock of the column) and load data again>(-55)
       MESSAGE
+      assert_equal(messages, error.error_output)
+      messages.split("\n").each do |message|
+        assert_includes(File.read(@log_path), message)
+      end
     end
 
     def test_force_truncate
       additional_path = "#{@column_path}.002"
       FileUtils.touch(additional_path)
-      result = grndb("recover", "--force-truncate")
-      assert_equal(<<-MESSAGE, result.error_output)
+      result = grndb("recover",
+                     "--force-truncate",
+                     "--log-level", "info")
+      messages = <<-MESSAGE
 [Users.age] Truncated broken object: <#{@column_path}>
 [Users.age] Removed broken object related file: <#{additional_path}>
       MESSAGE
+      assert_equal(messages, result.error_output)
+      messages.split("\n").each do |message|
+        assert_includes(File.read(@log_path), message)
+      end
     end
   end
 
```

Comment:
In my opinion, we should add "test_clear_lock" test case.

Reply:
Ah, I've got it.